### PR TITLE
Add Project.toml

### PR DESCRIPTION
**Pull request by rasmushenningsson**
# Add Project.toml

## Types of changes

This PR implements the following changes:

* [ ] :sparkles: New feature (A non-breaking change which adds functionality).
* [x] :bug: Bug fix (A non-breaking change, which fixes an issue).
* [ ] :boom: Breaking change (fix or feature that would cause existing functionality to change).

## :clipboard: Additional detail

- Updated from old REQUIRE to new Project.toml specification.
- Fixed some deprecation warnings.
- Updated travis and appveyor to run tests on Julia 1.5.

## :ballot_box_with_check: Checklist

- [x] :art: The changes implemented is consistent with the [julia style guide](https://docs.julialang.org/en/stable/manual/style-guide/).
- [x] :blue_book: I have updated and added relevant docstrings, in a manner consistent with the [documentation styleguide](https://docs.julialang.org/en/stable/manual/documentation/).
- [x] :blue_book: I have added or updated relevant user and developer manuals/documentation in `docs/src/`.
- [x] :ok: There are unit tests that cover the code changes I have made.
- [x] :ok: The unit tests cover my code changes AND they pass.
- [x] :pencil: I have added an entry to the `[UNRELEASED]` section of the manually curated `CHANGELOG.md` file for this repository.
- [x] :ok: All changes should be compatible with the latest stable version of Julia.
- [x] :thought_balloon: I have commented liberally for any complex pieces of internal code.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Automa = "67c07d97-cdcb-5c2c-af73-a7f9c32a568b"
+BGZFStreams = "28d598bf-9b8f-59f1-b38c-5a06b4a0f5e6"
+BioCore = "37cfa864-2cd6-5c12-ad9e-b6597d696c81"
+BioSequences = "7e6ae17a-c86d-528c-b3b9-7f778a29fe59"
+BufferedStreams = "e1450e63-4bb3-523b-b2a4-4ffa8c0fd77d"
+Combinatorics = "861a8166-3701-5b0c-9a16-15d98fcdc6aa"
+IntervalTrees = "524e6230-43b7-53ae-be76-1e9e4d08d11b"
+Twiddle = "7200193e-83a8-5a55-b20d-5d36d44a0795"
+
+[compat]
```
Comment (kescobo) [inline Project.toml:16]: How sure about these compat bounds are you? I suppose we can just set the bounds to current versions, and then test and relax things if anyone complains.

File: appveyor.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 environment:
   matrix:
-  - julia_version: 1
```
Comment (kescobo) [inline appveyor.yml:3]: If we're going to say it's compatible with 1.0, I think we should test on 1.0.

File: src/GeneticVariation.jl [hunk 3]
```diff
@@ -73,11 +73,9 @@ import IntervalTrees: Interval, IntervalValue
 import Twiddle:
     enumerate_nibbles,
     nibble_mask,
-    count_zero_nibbles,
+    count_0000_nibbles,
```
Comment (kescobo) [inline src/GeneticVariation.jl:76]: lol

File: Project.toml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+Automa = "0.8"
+BGZFStreams = "0.3"
+BioCore = "2.0.5"
```
Comment (kescobo) [inline Project.toml:19]: What's going on here? According to SemVer, this should be the same as 2.0, shouldn't it?

**Review by kescobo — COMMENTED** [review]
~~Any idea why there's such a massive coverage hit? Might be something about how coverage is calculated, I think I remember reading about that. In any case, there's nothing here that should affect it, so I don't think that should hold it up.~~

Just one minor change (add testing for julia 1.0) from me, and, if you have the energy, taking a look at compat bounds (though I'm fine with leaving those as-is until / unless someone complains).

I don't actually have merge rights on this repo, but looks good AFAICT

**Comment by kescobo** [comment]
> Any idea why there's such a massive coverage hit?

Ignore this - I see that's just the patch :+1:

File: Project.toml (hunk 1 above)
Comment (rasmushenningsson) [reply Project.toml:16]: I've checked and relaxed the compat bounds.
The biggest issue is incompatibility with BioSequences v2. But that requires larger changes and I think that's better to solve in another release. (And a better approach might be to split out a VCF.jl from this package that will simplify the compatibility situation.)

File: appveyor.yml (hunk 2 above)
Comment (rasmushenningsson) [reply appveyor.yml:3]: Agreed!

File: Project.toml (hunk 4 above)
Comment (rasmushenningsson) [reply Project.toml:19]: No, BioCore has some changes (bug fixes?) that we need. The tests won't pass unless we require at least v2.0.5.

File: Project.toml (hunk 1 above)
Comment (kescobo) [reply Project.toml:16]: Yeah, fixing that seems outside the scope of this PR.

**Comment by kescobo** [comment]
This LGTM - @BenJWard ?

**Comment by biona001** [comment]
Not to rush anyone, but will this PR be merged? I'm hoping this will solve a lot of the package compatibility issues I'm having with GeneticVariation.

**Comment by kescobo** [comment]
I would, but don't have permission. @BenJWard bump?

**Comment by jonathanBieler** [comment]
Would be great to have this merged yes, I just installed GeneticVariation and saw all my bio packages downgraded a major version or two :/

**Comment by rasmushenningsson** [comment]
I know it has been discussed before, so I decided to move the VCF/BCF functionality to a separate package - [VCF.jl](https://github.com/rasmushenningsson/VCF.jl). It's currently in the [process of being registered](https://github.com/JuliaRegistries/General/pull/30036).

**Comment by CiaranOMara** [comment]
Trusting the review, I'll merge and release this so that things may progress.

**Pull request merged**